02-express-tutorial/07-params-query.js

```diff
@@ -0,0 +1,58 @@
+const express = require('express');
+const app = express();
+const {products} = require('./data');
+
+app.get('/', (req, res) => {
+    res.send('<h1> Home Page</h1><a href ="/api/products">products</a>')
+})
+app.get('/api/products', (req,res) => {
+    const newProducts = products.map((product) => {
+        const { id, name, image } = product
+        return { id, name, image }
+    })
```

Comment:
There's another fun way to do this, by destructuring `product` right inside the function parameters, similar to how you would with require. I'll put some extra spaces to make it extra clear. You can see that this expression will print out the name for every product; feel free to try it out.

```javascript
products.forEach(  ({  name  }) => {
  console.log(name)
})
```

Similarly, we can pick out `id`, `name`, and `image` as you need to do here, then immediately return an object, achieving the same effect as what you've written. Again, I'll put extra spaces for clarity, because it gets pretty confusing with all those parentheses and curly braces.


```javascript
products.map(  ({ id, name, image }) => {
  return { id, name, image }
})
```

This can actually be further simplified:

```javascript
products.map(({id, name, image}) => ({id, name, image}))
```